src/utils/registry.ts

```diff
@@ -0,0 +1,58 @@
+import { detect } from 'package-manager-detector/detect';
+import { x } from 'tinyexec';
+
+export async function resolvePackageVersion(packageName: string): Promise<string> {
+	return fetchFromNpmRegistry(packageName);
+}
+
+async function fetchFromNpmRegistry(packageName: string): Promise<string> {
+	try {
+		const registry = await detectPackageRegistry();
+		const encodedName = encodeURIComponent(packageName).replace(/^%40/, '@');
+		// eslint-disable-next-line n/no-unsupported-features/node-builtins
```

Comment:
The fallback is [`>=16.0.0`](https://github.com/eslint-community/eslint-plugin-n/blob/17a60cd1abc8d70be6674f97521e8e98d5a415d1/lib/util/get-configured-node-version.js#L11).

i.e. add
```
    "n/no-unsupported-features/node-builtins": ["error", {
        "version": ">=18.12.0",
    }]
```
to `eslint.config.mjs` so that you can remove this comment.

see https://github.com/eslint-community/eslint-plugin-n/blob/17a60cd1abc8d70be6674f97521e8e98d5a415d1/lib/rules/no-unsupported-features/node-builtins.js#L39